### PR TITLE
fix(ingestion/airflow-plugin): bumping up the openlineage-airflow version

### PR DESCRIPTION
**Pull request by dushayntAW**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py [hunk 1]
```diff
@@ -419,12 +419,13 @@ def run_datajob(
         emit_templates: bool = True,
         config: Optional[DatahubLineageConfig] = None,
     ) -> DataProcessInstance:
-        if datajob is None:
+        if datajob is None and ti.task is not None:
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py:422]:
```suggestion
        if datajob is None:
            assert ti.task is not None
```

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py [hunk 2]
```diff
@@ -419,12 +419,13 @@ def run_datajob(
...
             datajob = AirflowGenerator.generate_datajob(
                 cluster, ti.task, dag, config=config
             )
 
         assert dag_run.run_id
+        assert datajob is not None
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py:428]: can be removed

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py [hunk 3]
```diff
@@ -508,7 +509,7 @@ def complete_datajob(
         :param config: Optional[DatahubLineageConfig]
         :return: DataProcessInstance
         """
-        if datajob is None:
+        if datajob is None and ti.task is not None:
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py:512]: same as above

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py [hunk 4]
```diff
@@ -376,113 +376,117 @@ def on_task_instance_running(
...
+                    for inlet in datajob.inlets:
+                        if not self.graph.exists(str(inlet)):
+                            logger.warning(f"Dataset {str(inlet)} not materialized")
 
     def on_task_instance_finish(
         self, task_instance: "TaskInstance", status: InstanceRunResult
     ) -> None:
         dagrun: "DagRun" = task_instance.dag_run  # type: ignore[attr-defined]
         task = self._task_holder.get_task(task_instance) or task_instance.task
-        dag: "DAG" = task.dag  # type: ignore[assignment]
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py:450]: add an `assert task is not None`, and remove the other changes

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py [hunk 5]
```diff
@@ -376,113 +377,118 @@ def on_task_instance_running(
...
-        for mcp in datajob.generate_mcp(
-            materialize_iolets=self.config.materialize_iolets
-        ):
-            self.emitter.emit(mcp, self._make_emit_callback())
-        logger.debug(f"Emitted DataHub Datajob finish w/ status {status}: {datajob}")
-
-        if self.config.capture_executions:
-            dpi = AirflowGenerator.complete_datajob(
-                emitter=self.emitter,
+        if task is not None:
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py:455]: No longer needed

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py [hunk 6]
```diff
@@ -376,113 +377,118 @@ def on_task_instance_running(
...
-        for mcp in datajob.generate_mcp(
-            materialize_iolets=self.config.materialize_iolets
-        ):
-            self.emitter.emit(mcp, self._make_emit_callback())
-        logger.debug(f"Emitted DataHub Datajob start: {datajob}")
-
-        if self.config.capture_executions:
-            dpi = AirflowGenerator.run_datajob(
-                emitter=self.emitter,
+        if task is not None:
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py:380]: No longer needed

**Comment by anshbansal** [comment]
Failure in nested domains is unrelated

**Pull request merged**